src/main/java/net/neoforged/neoforge/client/model/generators/ItemModelProvider.java

```diff
@@ -31,6 +32,33 @@ public ItemModelBuilder basicItem(ResourceLocation item) {
                 .texture("layer0", ResourceLocation.fromNamespaceAndPath(item.getNamespace(), "item/" + item.getPath()));
     }
 
+    public ItemModelBuilder handheldItem(Item item) {
+        return handheldItem(Objects.requireNonNull(BuiltInRegistries.ITEM.getKey(item)));
+    }
+
+    public ItemModelBuilder handheldItem(ResourceLocation item) {
+        return getBuilder(item.toString())
+                .parent(new ModelFile.UncheckedModelFile("item/handheld"))
+                .texture("layer0", ResourceLocation.fromNamespaceAndPath(item.getNamespace(), "item/" + item.getPath()));
+    }
+
+    public ItemModelBuilder spawnEggItem(Item item) {
+        return spawnEggItem(Objects.requireNonNull(BuiltInRegistries.ITEM.getKey(item)));
+    }
+
+    public ItemModelBuilder spawnEggItem(ResourceLocation item) {
+        return getBuilder(item.toString())
+                .parent(new ModelFile.UncheckedModelFile("item/template_spawn_egg"));
+    }
+
+    public ItemModelBuilder simpleBlockItem(Block block) {
+        return simpleBlockItem(Objects.requireNonNull(BuiltInRegistries.BLOCK.getKey(block)));
+    }
+
+    public ItemModelBuilder simpleBlockItem(ResourceLocation block) {
+        return withExistingParent(item.toString(), ResourceLocation.fromNamespaceAndPath(item.getNamespace(), "block/" + item.getPath()));
```

Comment:
The uses of the parameter need to be changed as well.
```suggestion
        return withExistingParent(block.toString(), ResourceLocation.fromNamespaceAndPath(block.getNamespace(), "block/" + block.getPath()));
```